iree/tools/android/run_module_app/build_apk.sh

```diff
@@ -0,0 +1,296 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+#########################
+# Script input parameters
+#########################
+
+echo "=== A script for building iree-run-module apk ==="
+echo "This script wraps iree-run-module together with a"
+echo "specific IREE VM module and invocation information."
+echo ""
+echo "At a minimum, it expects the following env vars:"
+echo "* ANDROID_SDK_ROOT"
+echo "* ANDROID_NDK_ROOT"
+echo "* ANDROID_SDK_BUILD_TOOLS_VERSION"
+echo "See the script for details and more controls."
+echo "===-------------------------------------------==="
+echo ""
+
+print_usage_and_exit() {
+  echo "Usage: $0 <artifact-directory> "
+  echo "       --driver <driver>"
+  echo "       --module_file <input-module-file> "
+  echo "       --entry_function <entry-function> "
+  echo "       --inputs_file <input-buffer-file> "
+  exit 1
+}
+
+while (( "$#" )); do
```

Comment:
Wow manual command line parsing in bash... I've heard that get-opt[s] is also maybe a good option?

Reply:
I was trying to provide the same command line options as other tools for consistency. I don't think getopts can parse options with double dash and multiple characters. But since both Scott and you dislike this, I'm fine changing it to getopts and just use single character as the options. :)

Reply:
I think that's a good goal. My "wow" was more like "wow I'm impressed" :grin: I've just avoided doing argument parsing in bash at all, sticking to env variables or breaking into python. getopt and getopts are two different tools for parsing, both of which have their issues. Honestly this isn't too bad with the manual way, if you think it will be maintainable